File: e-livraria-web/src/models/PricingGroup.ts
export interface IPricingGroup{
    id?: number;
    name?: string;
    discount?: number;
}
export class PricingGroup {
    private _id: number;
    private _name: string;
    private _discount: number;

    constructor({
        id = 0,
        name = '',
        discount = 0
    }: IPricingGroup) {
        this._id = id;
        this._name = name;
        this._discount = discount;
    }

    get id(): number {
        return this._id;
    }
    set id(id: number) {
        this._id = id;
    }

    get name(): string {
        return this._name;
    }
    set name(name: string) {
        this._name = name;
    }

    get discount(): number {
        return this._discount;
    }
    set discount(discount: number) {
        this._discount = discount;
    }
}

File: ecommerceLivraria/src/main/java/com/nrabello/ecommerceLivraria/model/PricingGroup.java
package com.nrabello.ecommerceLivraria.model;


import jakarta.persistence.*;

@Entity(name = "GrupoPrecificacao")
public class PricingGroup {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="grp_id")
    private Long id;
    @Column(name="grp_name")
    private String name;
    @Column(name="grp_porcentagem")
    private Double discount;

    public PricingGroup() {
    }

    public PricingGroup(Long id, String name, Double discount) {
        this.setId(id);
        this.setName(name);
        this.setDiscount(discount);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getDiscount() {
        return discount;
    }

    public void setDiscount(Double discount) {
        this.discount = discount;
    }

}
